pulp_smash/tests/ostree/api_v2/test_copy.py

```diff
@@ -0,0 +1,73 @@
+# coding=utf-8
+"""Copy content between OSTree repositories with a filter."""
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, config, utils
+from pulp_smash.constants import OSTREE_BRANCHES, OSTREE_FEED, REPOSITORY_PATH
+from pulp_smash.tests.ostree.utils import gen_distributor, gen_repo
+from pulp_smash.tests.ostree.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+class CopyTestCase(unittest.TestCase):
+    """Copy content between OSTree repositories with a filter."""
+
+    def test_all(self):
+        """Copy content between OSTree repositories with a filter..
+
+        Verify that:
+
+        Filters can be used to restrict which units
+        are copied between repositories.
+        """
+        repos = []  # list to store data related to created repos
+        cfg = config.get_config()
+        client = api.Client(cfg, api.json_handler)
+
+        # Create the 1st repository with at least 2 content units
+        body = gen_repo()
+        body['importer_config']['feed'] = OSTREE_FEED
+        body['importer_config']['branches'] = OSTREE_BRANCHES
+        body['distributors'].append(gen_distributor())
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        repos.append(repo)
+
+        # Sync the 1st repository
+        utils.sync_repo(cfg, repo)
+        search_repo = utils.search_units(cfg, repos[0])
+        first_unit_first_repo = search_repo[0]['metadata']['_id']
+        second_unit_first_repo = search_repo[1]['metadata']['_id']
+
+        # Create the 2nd repository
+        body = gen_repo()
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        repos.append(repo)
+
+        # Copy 2nd unit from the 1st repo to 2nd one.
+        client.post(urljoin(repos[1]['_href'], 'actions/associate/'), {
+            'source_repo_id': repos[0]['id'],
+            'criteria': {
+                'filters': {'unit': {'_id': second_unit_first_repo}},
+                'type_ids': ['ostree'],
+            },
+        })
+
+        # Assert that 1st unit was not copied into 2nd repo
+        criteria = {
+            'filters': {'unit': {'_id': first_unit_first_repo}},
+            'type_ids': ['ostree'],
+        }
+
+        len_search = len(utils.search_units(cfg, repos[1], criteria))
+        with self.subTest(comment='verify 1st unit was not copied'):
+            self.assertEqual(0, len_search)
+
+        # Search the 2nd unit on the 2nd repo
+        unit_second_repo = utils.search_units(cfg, repos[1])
+        unit_second_repo = unit_second_repo[0]['metadata']['_id']
+
+        # Assert that the 2nd unit was copied into 2nd repo
+        with self.subTest(comment='verify just filtered unit was copied'):
+            self.assertEqual(second_unit_first_repo, unit_second_repo)
```

Comment:
I think that the last 25 lines of code or so can be replaced by something like this:

```python
dst_unit_ids = [
    unit['metadata']['_id'] for unit in
    utils.search_units(cfg, repos[1], {'filters': {}, 'type_ids': ['ostree']})
]
self.assertEqual([src_unit_id], dst_unit_ids)
```